src/librustc/middle/lang_items.rs

```diff
@@ -327,6 +327,8 @@ language_item_table! {
     PanicBoundsCheckFnLangItem,      "panic_bounds_check",      panic_bounds_check_fn;
     PanicFmtLangItem,                "panic_fmt",               panic_fmt;
 
+    // ExchangeMallocFnLangItem cannot unwind, or MIR trans will break. See note
+    // on `malloc_raw_dyn` in librustc_trans/base.rs.
```

Comment:
Good to put this into liballoc. Is this a new requirement, or just a new comment? Since we know the discussion about panic on oom will come back.